src/video/SDL_egl.c

```diff
@@ -511,6 +512,22 @@ SDL_EGL_LoadLibrary(_THIS, const char *egl_path, NativeDisplayType native_displa
          * Therefore SDL_EGL_GetVersion() shouldn't work with uninitialized display.
          * - it actually doesn't work on Android that has 1.5 egl client
          * - it works on desktop X11 (using SDL_VIDEO_X11_FORCE_EGL=1) */
+#if defined(SDL_VIDEO_DRIVER_ROCKCHIP)
+	if (SDL_EGL_HasExtension(_this, SDL_EGL_CLIENT_EXTENSION, "EGL_EXT_platform_base")) {
+	    _this->egl_data->eglGetPlatformDisplayEXT = SDL_EGL_GetProcAddress(_this, "eglGetPlatformDisplayEXT");
+	     if (_this->egl_data->eglGetPlatformDisplayEXT) {
+		 _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplayEXT(platform, native_display, NULL);
+	     } else {
+		 if (_this->egl_data->eglGetPlatformDisplay) {
+		     _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplay(platform, native_display, NULL);
+		  }
+	     }
+	} else {
+	     if (_this->egl_data->eglGetPlatformDisplay) {
+		 _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplay(platform, native_display, NULL);
+	     }
+	}
+#else
```

Comment:
What is wrong with the existing code here on Rockchip?

Reply:
Well, it is a hack in order to make it work as expected.

Reply:
Yes, but why is the existing code failing? What error is occurring?

Reply:
I can't remember the exact error, i think the 0x3003 error. must be the mali blob implementation.

Reply:
I still don't understand which functions are being called and which are failing. Can you please describe specifically what is going wrong with the existing code and how your changes fix it?

It doesn't look like the `eglGetPlatformDisplay()` codepaths are even reachable in the Rockchip codepath, since you don't call `LOAD_FUNC(eglGetPlatformDisplay)`.

Reply:
The path that works is this one:

```
if (SDL_EGL_HasExtension(_this, SDL_EGL_CLIENT_EXTENSION, "EGL_EXT_platform_base")) {
	    _this->egl_data->eglGetPlatformDisplayEXT = SDL_EGL_GetProcAddress(_this, "eglGetPlatformDisplayEXT");
	     if (_this->egl_data->eglGetPlatformDisplayEXT) {
		 _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplayEXT(platform, native_display, NULL);
```

Reply:
> what is going wrong with the existing code and how your changes fix it?

_this->egl_data->egl_display is not created, using EXTENSION creates the egl_display correctly.

Reply:
OK, in that case, we can simplify this to something that works for Rockchip and other implementations.

Does it work properly if you remove your Rockchip codepath and then change this
```
        if (_this->egl_data->eglGetPlatformDisplay) {
            _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplay(platform, (void *)(uintptr_t)native_display, NULL);
        } else {
            if (SDL_EGL_HasExtension(_this, SDL_EGL_CLIENT_EXTENSION, "EGL_EXT_platform_base")) {
                _this->egl_data->eglGetPlatformDisplayEXT = SDL_EGL_GetProcAddress(_this, "eglGetPlatformDisplayEXT");
                if (_this->egl_data->eglGetPlatformDisplayEXT) {
                    _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplayEXT(platform, (void *)(uintptr_t)native_display, NULL);
                }
            }
        }
```

to this?

```
        if (_this->egl_data->eglGetPlatformDisplay) {
            _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplay(platform, (void *)(uintptr_t)native_display, NULL);
        }

        if (_this->egl_data->egl_display == EGL_NO_DISPLAY && SDL_EGL_HasExtension(_this, SDL_EGL_CLIENT_EXTENSION, "EGL_EXT_platform_base")) {
            _this->egl_data->eglGetPlatformDisplayEXT = SDL_EGL_GetProcAddress(_this, "eglGetPlatformDisplayEXT");
            if (_this->egl_data->eglGetPlatformDisplayEXT) {
                 _this->egl_data->egl_display = _this->egl_data->eglGetPlatformDisplayEXT(platform, (void *)(uintptr_t)native_display, NULL);
            }
        }
```

Reply:
Probably yes, going to test it...

Reply:
Please, review the changes, according to my annotations here: https://github.com/libsdl-org/SDL/issues/4879#issuecomment-963305796